### PR TITLE
Fix dashboard menu styles

### PR DESCRIPTION
**Pull request by benekex2**
I want to merge this change because...

### Pull Request Checklist

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Database queries are optimized and the number of queries is constant.
1. [ ] Database migration files are up to date.
1. [ ] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).
1. [ ] GraphQL schema and type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### CODE REVIEW

File: saleor/static/dashboard-next/components/AppLayout/AppLayout.tsx [hunk 1]
```diff
@@ -91,16 +91,29 @@ const styles = (theme: Theme) =>
...
+      },
       background: theme.palette.background.paper,
+      border: `solid 1px #EAEAEA`,
       borderRadius: "50%",
       cursor: "pointer",
       height: 32,
       position: "absolute",
       right: -16,
       top: 65,
+      transition: "background 0.2s",
```
Comment (maarcingebala) [inline saleor/static/dashboard-next/components/AppLayout/AppLayout.tsx]: Shouldn't we use a variable for the transition time value? It's used in a few places and I guess we always want to use the same value for all of them?

File: saleor/static/dashboard-next/theme.ts [hunk 2]
```diff
@@ -328,6 +328,11 @@ export default (colors: IThemeColors): Theme =>
       createShadow(9, 44, 7, 11, 16, -6, 22, 35, 2),
       createShadow(9, 46, 8, 11, 15, -7, 24, 38, 3)
     ],
+    transitions: {
+      duration: {
+        short: 0.2
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/theme.ts]: You could just use `shorter`, it's set to `200`

File: saleor/static/dashboard-next/components/AppLayout/AppLayout.tsx [hunk 3]
```diff
@@ -91,16 +91,29 @@ const styles = (theme: Theme) =>
...
       right: -16,
       top: 65,
+      transition: `background ${theme.transitions.duration.short}s`,
       width: 32,
       zIndex: 99
     },
+    isMenuSmallDark: {
+      "&:hover": {
+        background:
+          "linear-gradient(0deg, rgba(25, 195, 190, 0.1), rgba(25, 195, 190, 0.1)), #2E2F31"
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/components/AppLayout/AppLayout.tsx]: `#2e2f31` is `theme.palette.background.paper`

File: saleor/static/dashboard-next/components/AppLayout/AppLayout.tsx [hunk 4]
```diff
@@ -248,7 +259,7 @@ interface AppLayoutProps {
 const AppLayout = withStyles(styles, {
   name: "AppLayout"
 })(
-  withRouter<AppLayoutProps & RouteComponentProps<any>, any>(
+  withRouter<AppLayoutProps & RouteComponentProps<any>>(
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/components/AppLayout/AppLayout.tsx]: Does it build now?

**Comment by maarcingebala** [comment]
@benekex2 CI is still failing here

**Comment by django-queries** [comment]
Here is the report for 962ac96dcfa81bdb8c1ce8b69491ff9d0e0a3e09 (mirumee/saleor @ fix/dashboard-menu-styles)
Base comparison is abfcf33e172d68c342bf42c75d6171213c13d567.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  add billing address to checkout     	         34	         34	             20
  add shipping to checkout            	          7	          7	              0
  checkout payment charge             	         14	         14	              0
  complete checkout                   	          6	          6	              0
  create checkout                     	         43	         43	             21

# api.benchmark homepage
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve main menu                  	          5	          5	              0
  retrieve product list               	          4	          4	              0
  retrieve secondary menu             	          5	          5	              0
  retrieve shop                       	          2	          2	              0

# api.benchmark product
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  product details                     	         13	         13	              3

# api.benchmark variant
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve variant list               	          9	          9	              2
```

</p>
</details>

**Pull request merged**